### PR TITLE
MHV-64869: Self-entered pdf

### PR DESCRIPTION
**Pull request by mattwrightva**
Did you change site-wide styles, platform utilities or other infrastructure?
- [ ] No
- [x] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
- Self-entered data pdf generation completed. 
- Blue button pdf fixes implemented
- Minor PDF fixes and enhancements
- Removed table of contents from Blue Button PDF

## Related issue(s)
https://jira.devops.va.gov/browse/MHV-64869
> ### Update the self entered records PDF downloads with live data
> Description
> Update the self entered records PDF downloads with live data
> 

## Testing done
Unit tests fixed where necessary. 

## Screenshots
Self-entered PDF:
Screenshot 2024-12-09 at 3 52 40 PM
Screenshot 2024-12-09 at 3 52 24 PM
Screenshot 2024-12-09 at 3 52 15 PM
Screenshot 2024-12-09 at 3 52 08 PM
Screenshot 2024-12-09 at 3 52 02 PM
Screenshot 2024-12-09 at 3 51 56 PM
Screenshot 2024-12-09 at 3 51 49 PM
Screenshot 2024-12-09 at 3 51 40 PM
Screenshot 2024-12-09 at 3 51 31 PM
Screenshot 2024-12-09 at 3 51 20 PM
Screenshot 2024-12-09 at 3 51 10 PM
Screenshot 2024-12-09 at 3 50 50 PM
Screenshot 2024-12-09 at 3 50 42 PM
Screenshot 2024-12-09 at 3 50 34 PM
Screenshot 2024-12-09 at 3 50 18 PM

Blue button fixes:
Screenshot 2024-12-09 at 3 57 16 PM
Screenshot 2024-12-09 at 3 56 56 PM
Screenshot 2024-12-09 at 3 56 50 PM
Screenshot 2024-12-09 at 3 53 22 PM
Screenshot 2024-12-09 at 3 53 04 PM

## What areas of the site does it impact?
MHV medical records

## Acceptance criteria
The self entered records PDF and TXT downloads have been updated with live data

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/mhv-medical-records/components/LabsAndTests/PathologyDetails.jsx [hunk 1]
```diff
@@ -114,12 +121,24 @@ ${record.results} \n`;
 
       <div className="test-details-container max-80">
         <h2>Details about this test</h2>
+        <h3 className="vads-u-font-size--md vads-u-font-family--sans">
+          Date and time collected
+        </h3>
+        <p data-testid="pathology-date" data-dd-privacy="mask">
+          {record.date}
+        </p>
```
Comment (mmoyer-va) [inline src/applications/mhv-medical-records/components/LabsAndTests/PathologyDetails.jsx:129]: The `DateSubheading` is also printing "Date and time collected". That one uses `record.dateCollected` and this one uses `record.date`. What is the difference between these? Which is correct? Should the field be duplicated?

File: src/applications/mhv-medical-records/components/LabsAndTests/MicroDetails.jsx [hunk 2]
```diff
@@ -125,16 +125,12 @@ ${record.results}`;
...
-              {record.labType}
-            </p>
-          </>
-        )}
+        <h3 className="vads-u-font-size--md vads-u-font-family--sans">
+          Date and time collected
+        </h3>
+        <p data-testid="microbio-date-completed" data-dd-privacy="mask">
+          {record.date}
+        </p>
```
Comment (mmoyer-va) [inline src/applications/mhv-medical-records/components/LabsAndTests/MicroDetails.jsx:133]: This field is duplicated exactly in the `DateSubheading` above.

**Review by mmoyer-va — CHANGES_REQUESTED** [review]

File: src/applications/mhv-medical-records/components/LabsAndTests/PathologyDetails.jsx (hunk 1 above)
Comment (mattwrightva) [reply src/applications/mhv-medical-records/components/LabsAndTests/PathologyDetails.jsx:129]: Good catch! I added it after adding it to the PDF but didn't consider it was also above.

File: src/applications/mhv-medical-records/components/LabsAndTests/MicroDetails.jsx (hunk 2 above)
Comment (mattwrightva) [reply src/applications/mhv-medical-records/components/LabsAndTests/MicroDetails.jsx]: Good catch!

**Review by mmoyer-va — DISMISSED** [review]

**Review by mmoyer-va — DISMISSED** [review]

**Review by CBonade — DISMISSED** [review]

**Review by mmoyer-va — DISMISSED** [review]

**Review by mmoyer-va — APPROVED** [review]

**Review by CBonade — APPROVED** [review]

**Pull request merged**